src/store/messagesStore.js

```diff
@@ -1256,30 +1259,60 @@ const actions = {
 	},
 
 	/**
-	 * Posts a simple text message to a room
+	 * Forwards message to a conversation. By default , the message is forwarded to Note to self.
 	 *
 	 * @param {object} context default store context;
 	 * will be forwarded;
 	 * @param {object} data the wrapping object;
+	 * @param {object} data.targetToken the conversation token to where the message will be forwarded;
 	 * @param {object} data.messageToBeForwarded the message object;
 	 */
-	async forwardMessage(context, { messageToBeForwarded }) {
-		const response = await postNewMessage(messageToBeForwarded, { silent: false })
-		return response
-	},
+	async forwardMessage(context, { targetToken, messageToBeForwarded }) {
+		const message = cloneDeep(messageToBeForwarded)
+
+		// when there is no token provided, the message will be forwarded to the Note to self conversation
+		if (!targetToken) {
+			let noteToSelf = context.getters.conversationsList.find(conversation => conversation.type === CONVERSATION.TYPE.NOTE_TO_SELF)
+			// If Note to self doesn't exist, it will be regenerated
+			if (!noteToSelf) {
+				const response = await fetchNoteToSelfConversation()
+				noteToSelf = response.data.ocs.data
+				context.dispatch('addConversation', noteToSelf)
+			}
+			targetToken = noteToSelf.token
+		}
+		// Overwrite with the target conversation token
+		message.token = targetToken
+		if (message.parent) {
+			delete message.parent
+		}
+
+		if (message.message === '{object}' && message.messageParameters.object) {
+			const richObject = message.messageParameters.object
+			const response = await postRichObjectToConversation(
+				targetToken,
+				{
+					objectId: richObject.id,
+					objectType: richObject.type,
+					metaData: JSON.stringify(richObject),
+					referenceId: '',
+				},
+			)
+			return response
+		}
 
-	/**
-	 * Posts a simple text message to a room
-	 *
-	 * @param {object} context default store context;
-	 * will be forwarded;
-	 * @param {object} data the wrapping object;
-	 * @param {string} data.token token of the target conversation
-	 * @param {object} data.richObject the rich object;
-	 */
-	async forwardRichObject(context, { token, richObject }) {
-		const response = await postRichObjectToConversation(token, richObject)
+		// If there are mentions in the message to be forwarded, replace them in the message
+		// text.
+		for (const key in message.messageParameters) {
+			if (key.startsWith('mention')) {
+				const mention = message.messageParameters[key]
+				message.message = message.message.replace(`{${key}}`, `@'${mention.name}'`)
+			}
+		}
+
+		const response = await postNewMessage(message, { silent: true })
```

Comment:
Used to be false, now it's true, meaning if you forward something to a 1-1, the other participant will not be notified.

Unless we add an option for the user to select it, I would stand with above comment also here:
We can have a look into the handling later, but let's not mix it with this "must merge" feature request.
Keep the previous behaviour for now.

```suggestion
		const response = await postNewMessage(message, { silent: false })
```